exchanges/graphcache/src/cacheExchange.ts

```diff
@@ -182,7 +183,7 @@ export const cacheExchange =
         operation.kind,
         {
           key: operation.key,
-          query: formatDocument(operation.query),
+          query: formatDocument(operation.query, directivesToRemove),
```

Comment:
This won't remove the document as it's cached in `formatDocument`, would it be okay to remove the cache or bust it here by removing `operation.query.__key`? An alternative would have been to use i.e. `_` as a prefix to graphcache-directives but then we'd have to store them on the AST after formatting the document as we would lose it too early on in our chain.